src/components/project-input.ts

```diff
@@ -1,74 +1,77 @@
+  descriptionInputElement: HTMLInputElement;
+  peopleInputElement: HTMLInputElement;
+  constructor() {
+    //super(the id of the template we gonna use,where we gonna store the values,if we want to add at start(true) else at the
+    //end,if we want to add an Id to the element we are creating)
+    super("project-input", "app", true, "user-input");
+    this.titleInputElement = this.element.querySelector(
+      "#title"
+    )! as HTMLInputElement;
+    this.descriptionInputElement = this.element.querySelector(
+      "#description"
+    )! as HTMLInputElement;
+    this.peopleInputElement = this.element.querySelector(
+      "#people"
+    )! as HTMLInputElement;
+    this.configure();
+  }
+  configure() {
+    this.element.addEventListener("submit", this.submitHandler);
+  }
+  renderContent(): void {}
+  //on submit
+  @autobind
+  private submitHandler(event: Event): void {
+    event.preventDefault();
+    const userInput = this.gatherUserInput();
+    //if user input is correct we receive the array
+    if (Array.isArray(userInput)) {
+      const [title, description, people] = userInput;
+      projectState.addProject(title, description, people);
+      this.clearUserInputs();
     }
-    private gatherUserInput(): [string, string, number] | void {
-      const enteredTitle = this.titleInputElement.value;
-      const enteredDescription = this.descriptionInputElement.value;
-      const enteredPeoples = this.peopleInputElement.value;
-      const titleValidatable: Validatable = {
-        value: enteredTitle,
-        required: true,
-      };
-      const descriptionValidatable: Validatable = {
-        value: enteredDescription,
-        required: true,
-        minLength: 7,
-      };
-      const peopleValidatable: Validatable = {
-        value: +enteredPeoples,
-        required: true,
-        minValue: 1,
-      };
-      if (
-        !validate(titleValidatable) ||
-        !validate(descriptionValidatable) ||
-        !validate(peopleValidatable)
-      ) {
-        alert("Please enter valid input");
-      } else {
-        return [enteredTitle, enteredDescription, Number(enteredPeoples)];
-      }
+  }
+  //cleans user input fields
+  private clearUserInputs() {
+    this.titleInputElement.value = "";
+    this.descriptionInputElement.value = "";
+    this.peopleInputElement.value = "";
+  }
+  //gets and validates user input
+  private gatherUserInput(): [string, string, number] | void {
+    const enteredTitle = this.titleInputElement.value;
+    const enteredDescription = this.descriptionInputElement.value;
+    const enteredPeoples = this.peopleInputElement.value;
+    //adds the requirements for each input and then calls the validate function to check if they meet the requirements
+    const titleValidatable: Validatable = {
+      value: enteredTitle,
+      required: true,
+    };
+    const descriptionValidatable: Validatable = {
+      value: enteredDescription,
+      required: true,
+      minLength: 7,
```

Comment:
when you showed me the project we had error that we didin't meet the min length requirement, but it didn't say what we didn't meet. IF that is still the case, can you point out in the alert what is the requirement that hasn't been met so the user knows what to do